lib/srv/db/postgres/engine.go

```diff
@@ -107,6 +113,10 @@ func (e *Engine) HandleConnection(ctx context.Context, sessionCtx *common.Sessio
 	if err != nil {
 		return trace.Wrap(err)
 	}
+	if e.cancelReq != nil {
+		// Special case when sending a cancel request.
+		return e.handleCancelRequest(ctx, sessionCtx)
```

Comment:
no audit log event. Since cancel requests establish a new connection, the only way to associate a cancel request with a session is by adding PID to the session as well, and there's no way to know if a cancel request actually cancelled anything either. And then every time the user hits `ctrl+c`, there will be audit log events for a new session start, cancel request sent, session end. I feel like there were other problems I'm forgetting that came up when discussing this with @smallinsky last week, but we opted to skip audit logging changes at least for now.